src/test/ui/consts/issue-88649.rs

```diff
@@ -0,0 +1,18 @@
+// check-pass
+#![crate_type = "lib"]
+
+enum Foo {
+    Variant1(bool),
+    Variant2(bool),
+}
+
+const _: () = {
+    let mut n = 0;
+    while n < 2 {
+        match Foo::Variant1(true) {
+            Foo::Variant1(x) | Foo::Variant2(x) if x => {}
+            _ => {}
+        }
+        n += 1;
+    }
+};
```

Comment:
Why did you put it in a `const`? There's a simpler version of this that failed before #88678 and succeeds after:

```rust
fn main() {
    loop {
        match Foo::Variant1(true) {
            Foo::Variant1(x) | Foo::Variant2(x) if x => {}
            _ => {}
        }
    }
}
```

(From #88649.)

Reply:
I think that requires Miri to run? It does not seem to fail in `rustc`, so I don't think it works in a UI test.

Reply:
Ah, I see; you put it in a `const` so that the rustc version of Miri for const-eval would run 👍 

I think you could still simplify the `while` into an unqualified `loop`, but your call.

Reply:
> I think you could still simplify the `while` into an unqualified `loop`, but your call.

Using an unqualified `loop` means that it will enter an infinite loop and fails with this:
```
9  | / const _: () = {
10 | |     loop {
11 | |         match Foo::Variant1(true) {
12 | |             Foo::Variant1(x) | Foo::Variant2(x) if x => {}
   | |                                                          ^ exceeded interpreter step limit (see `#[const_eval_limit]`)
...  |
15 | |     }
16 | | };
   | |__-
```
I think I can test ``//~ ERROR exceeded interpreter step limit (see `#[const_eval_limit]`)`` instead, but I think it may be clearer to make this a `check-pass` test without an infinite loop.
(However, if `while n < 2` is actually more confusing, I'm fine with changing this to `loop`.)

Reply:
That makes a lot of sense. Sorry for the noise :)